src/main/java/uk/gov/ida/matchingserviceadapter/mappers/MatchingDatasetToMatchingDatasetDtoMapper.java

```diff
@@ -18,14 +18,15 @@
 import uk.gov.ida.saml.core.domain.TransliterableMdsValue;
 
 import javax.annotation.Nullable;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 public class MatchingDatasetToMatchingDatasetDtoMapper {
 
     public VerifyMatchingDatasetDto mapToVerifyMatchingDatasetDto(MatchingDataset matchingDataset) {
-        Optional<TransliterableMdsValue> firstNameValue = matchingDataset.getFirstNames().stream().findFirst();
+        Optional<TransliterableMdsValue> firstNameValue = matchingDataset.getFirstNames().stream().min(getFirstNameComparator());
         Optional<SimpleMdsValue<String>> middleNameValue = matchingDataset.getMiddleNames().stream().findFirst();
         Optional<SimpleMdsValue<LocalDate>> birthDateValue = matchingDataset.getDateOfBirths().stream().findFirst();
 
```

Comment:
We should probably do the same comparison for middleName and DoB.

Reply:
That needs a larger discussion before I'd be willing to make any changes. The card - and the original zendesk ticket - have only ever been about the first names. (In total agreement it makes sense to use this logic on the other fields; I'm just not familiar enough with how things work to do it here.)

Can this be raised as an issue or ticket (or whatever the process is) for discussion?

Reply:
OK, Definitely worth raising another ticket in that case.

With reference to the ticket and zendesk, we should put in an integration test that exercises the original failing validation. e.g. multiple current first names.

Reply:
Created a [new Trello card](https://trello.com/c/cfAXx6BU/) for the follow-up.

Also 🤦‍♂️ I had an integration test and seem to have not included it in my commits. 👍 for pointing that out.

Reply:
Integration test added. 👍